runtime/contract_function_executor.go

```diff
@@ -183,10 +183,12 @@ func (executor *interpreterContractFunctionExecutor) execute() (val cadence.Valu
 		return nil, newError(err, location, codesAndPrograms)
 	}
 
+	var self interpreter.MemberAccessibleValue = contractValue
+
 	// prepare invocation
 	invocation := interpreter.NewInvocation(
 		inter,
-		contractValue,
+		&self,
```

Comment:
So am I correct then in understanding that this would now be a double-pointer `**CompositeValue` type in the non-`nil` case for `contractValue`? Getting flashbacks to C++ here

Reply:
Correct